From 5a01450659fa329157d6996f10d4a2ad2ad2decd Mon Sep 17 00:00:00 2001
From: Alison Gale <agale@chromium.org>
Date: Fri, 16 Jan 2026 18:35:05 -0800
Subject: [PATCH] [Vertical Tabs] Disable scroll to change tabs on Linux

Bug: 474165324
Change-Id: I71eff12cf6f50c238cfbf72d5f0c7789037f5ba0
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/7486711
Commit-Queue: Alison Gale <agale@chromium.org>
Reviewed-by: Vince Lugli <lugli@google.com>
Cr-Commit-Position: refs/heads/main@{#1570728}
---

--- a/chrome/browser/ui/views/frame/browser_root_view.cc
+++ b/chrome/browser/ui/views/frame/browser_root_view.cc
@@ -329,9 +329,10 @@ bool BrowserRootView::OnMouseWheel(const
   // TODO(dfried): See if it's possible to move this logic deeper into the view
   // hierarchy - ideally to HorizontalTabStripRegionView.
 
-  // Scroll-event-changes-tab is incompatible with scrolling tabstrip, so
+  // Scroll-event-changes-tab is incompatible with vertical tabstrip, so
   // disable it if the latter feature is enabled.
-  if (browser_defaults::kScrollEventChangesTab) {
+  if (browser_defaults::kScrollEventChangesTab &&
+      !browser_view_->ShouldDrawVerticalTabStrip()) {
     // Switch to the left/right tab if the wheel-scroll happens over the
     // tabstrip, or the empty space beside the tabstrip.
     views::View* hit_view = GetEventHandlerForPoint(event.location());
